### ISSUE TITLE
Repo readme / NPM release / current status?

### ISSUE THREAD

**Issue by markerikson** [davidkpiano/redux-simple-form#1]
This looks interesting.  However, it's a bit hard to check it out right now, because the only NPM release appears to be out-of-date, and there's no readme.  I do see that you've got actual documentation on GH Pages, which is certainly a plus.  What's the current status of this library?

**Comment by davidkpiano** [comment]
@markerikson Funny you discovered this, I haven't released it yet. Right now I'm working on unit tests, and I want to write as many unit tests as possible before this gets released. But if you're curious, here's the current state of documentation: http://davidkpiano.github.io/redux-simple-form/

**Comment by markerikson** [comment]
Yeah, I keep searching through Github for newly updated repos with "redux" in the name.  Found some interesting items that way, too.

I'm currently doing prototyping for my first Redux project, a rewrite of an application that among other things involves a bunch of form editing.  That's next on my list of things to figure out, so I've got a bunch of different React/Redux-related form libs I'm going to be looking at.  redux-form is certainly the big name out there, but at first glance it seems a bit bigger and over-engineered than I'd like, so I'm looking for various other options to try out.

I'll keep an eye on this to see how it develops.

**Comment by davidkpiano** [comment]
So my goals with **redux-simple-form** are:

1. Make it conceptually simple (I agree, `redux-form` is too API-heavy, proprietary, and over-engineered)
2. Model the way Angular does forms (because, you have to admit, forms in Angular are a breeze)
3. Performance (only validate changed fields rather than every field, give ability to only change model on blur, debounce, etc.)
4. **Keep your models/stores intact.** This is huge - I don't want my models/stores to be mixed into my form data, and I want to take control of my model as much as possible.

What do you think? Anything else you'd like to see in a Redux-form library?

**Comment by markerikson** [comment]
Never touched Angular, so I have no point of comparison there.  

Other than that, sounds reasonable.  

I'll go ahead and toss out my use case, just for reference.  My application involves a bunch of flight planning data, so lots of things like lat/lon/alt waypoints, etc.  I have some forms for viewing attributes of an item (name, location, some various enums and booleans and numbers), as well as a form for editing a list of waypoints.  My old app had some custom composite widgets for editing a lat/lon (textbox with a bunch of validation logic), an altitude (numeric input plus a dropdown), and a position (lat/lon + altitude combined).  The waypoint list allows adding/removing/rearranging entries, and the whole parent form can reset to the last saved value or apply the in-form edits to the actual value.

The original was written in GWT+SmartGWT several years back, when I only sort of knew what I was doing, so I'm hoping to do things a lot better this time around :)

**Comment by davidkpiano** [comment]
@markerikson I'd love to see what your form looks like visually, for reference! I want to add a bunch of examples and dog-food this as much as I can before I release it to the public.

**Comment by markerikson** [comment]
Side question, because I don't want to open up a whole new issue: is the explicit dependency on "react@^0.14.6" a problem?  The boilerplate I started with is still set to "^0.14.0", and when I tried doing an NPM install for this repo, it pulled down 0.14.6 as a nested dep.  Still learning my way around the NPM ecosystem, but seems like that'd introduce a second copy of React into the build, which would be bad.  Does it maybe need to be a "peerDependency" or something?
